lib/sherlock_homes/property_finder.rb

```diff
@@ -0,0 +1,25 @@
+module SherlockHomes
+
+  class ZillowServiceError < RuntimeError; end
+  class AddressResolveError < RuntimeError; end
+  class NoPropertyError < RuntimeError; end
+
+  class PropertyFinder
+
+    def initialize
+      Rubillow.configure { |c| c.zwsid = SherlockHomes.configuration.zillow_key }
+    end
+
+    def search(address: '', citystatezip: '')
+      property = Rubillow::PropertyDetails.deep_search_results(address: address, citystatezip: citystatezip)
```

Comment:
Let's validate that both an address, and citystatezip are provided, and raise before running the search otherwise.